patch_parser_test.py

```diff
@@ -1,11 +1,16 @@
 import unittest
 from patch_parser import parse_comments
-from archive_converter import find_thread, generate_email_from_file
+from archive_converter import ArchiveMessageIndex, generate_email_from_file
+from message_dao import MessageDao
 
 class PatchParserTest(unittest.TestCase):
 
     def test_parse_comments_for_single_email_thread(self):
-        patchset = parse_comments(find_thread('PATCH v2 1/2'))
+        archive_index = ArchiveMessageIndex(MessageDao())
+        archive_index.update('test_data')
+        patchset = parse_comments(
+            archive_index.find(
+                '<20200827144112.v2.1.I6981f9a9f0c12e60f8038f3b574184f8ffc1b9b5@changeid>'))
```

Comment:
can we include an assert on the subject or similar?
Would make the test a bit easier to read, imo.

This seems to be patch6.txt, aka
> Subject: [PATCH v2 1/2] Input: i8042 - Prevent intermixing i8042 commands

Reply:
i'm not too sure what this means. I assert on the subject later in that test, should I be asserting on the find? like self.assertEquals(archive_index.find('hash').subject, "Subject: [PATCH v2 1/2] Input: i8042 - Prevent intermixing i8042 commands"))

Reply:
I just meant adding something like

```python
self.assertEqual(first_patch.subject, 'our subject')
```

The goal is to just make it more clear to the test reader what `first_patch` is expected to be.
A patch with no comments on it isn't much to go on.
It'd be easier to go grep for a few words from the subject then to copy-paste the message id (e.g. it might be unclear if the message id will appear exactly in the same form in the testdata files)

Reply:
I thought the patches didn't have a subject field? Once they transform from message to patches, I don't think they have the same fields. I try to do the subject comparison by doing the self.assertIn on line 19. Should I comment that it's the subject comparison?

Reply:
Ah yes, you're right.
I didn't notice that GitHub hid 3 lines from me which had the `self.assertIn` I was asking you to write.

(Why would it collapse just those few lines? 😞 )